packages/flutter/lib/src/widgets/animated_size.dart

```diff
@@ -49,21 +49,27 @@ import 'framework.dart';
 /// See also:
 ///
 ///  * [SizeTransition], which changes its size based on an [Animation].
-class AnimatedSize extends SingleChildRenderObjectWidget {
+class AnimatedSize extends StatefulWidget {
   /// Creates a widget that animates its size to match that of its child.
   ///
   /// The [curve] and [duration] arguments must not be null.
   const AnimatedSize({
     Key? key,
-    Widget? child,
+    this.child,
     this.alignment = Alignment.center,
     this.curve = Curves.linear,
     required this.duration,
     this.reverseDuration,
-    required this.vsync,
+    // This field is now ignored and will be deprecated.
```

Comment:
Can you phrase this as a "TODO(xx): deprecate when customer tests are updated"?

Reply:
done